control-plane/agents/core/src/pool/specs.rs

```diff
@@ -59,6 +65,14 @@ impl ResourceSpecs {
         }
         replicas
     }
+    pub(crate) async fn get_replicas(&self) -> Vec<ReplicaSpec> {
+        let mut vector = vec![];
+        for object in self.replicas.values() {
```

Comment:
Do we require a lock on the replicas to ensure the hash map is not being modified whilst we are iteration over it?

Reply:
As discussed in a call, we're already locked, will update with comments accordingly